wrappers/src/object_cs.cpp

```diff
@@ -241,11 +241,18 @@ extern "C" {
         });
     }
 
-    REALM_EXPORT int64_t object_get_obj_key(const Object& object, NativeException::Marshallable& ex)
+    REALM_EXPORT int32_t object_get_hashcode(const Object& object, NativeException::Marshallable& ex)
     {
         return handle_errors(ex, [&]() {
-            // ObjKey is incompatible with C, so we return just the value.
-            return object.obj().get_key().value;
+            auto table_key_value = object.obj().get_table()->get_key().value;
+            auto table_key_hash = std::hash<int>{}(table_key_value);
```

Comment:
Do we need these `std::hash`-es here? Can we use `table_key_value` directly - maybe with a cast?

Reply:
I guess the normal way would be to only use them since they are already provided by the standard libary but the other way round should be fine as well. Removed them.

Reply:
Yeah, I'm not familiar with how C++ does things, so happy to use the idiomatic way there - it's just coming from C#, you would rarely use `GetHashCode` on a primitive type when you can just use the value itself.

Reply:
I did some more research now that you mention it. Good point about the primitives:
https://stackoverflow.com/questions/38304877/why-stdhashint-seems-to-be-identity-function
It makes sense but means that I can just remove it since `std::hash<int>(i) == i` anyway.

I'll just keep it our way then. There are certainly many more ways but if the current implementation is fine I guess we can just keep it.
I thought about just concatenating table key and object key since that would always lead to a unique 'hash':
```
auto table_key_value = object.obj().get_table()->get_key().value;
auto object_key_value = object.obj().get_key().value;

std::string concatenation = "" + table_key_value + object_key_value;
int hashCode = std::stoi(concatenation);

return hashCode;
```